src/gt4py/gtc/python/oir_to_npir.py

```diff
@@ -0,0 +1,104 @@
+from typing import Dict, List, Optional
+
+from dataclasses import dataclass, field
+from eve.visitors import NodeTranslator
+
+from .. import common, oir
+from . import npir
+
+
+class OirToNpir(NodeTranslator):
+    """Lower from optimizable IR (OIR) to numpy IR (NPIR)."""
+
+    @dataclass
+    class Context:
+        """Context for a HorizontalExecution."""
+
+        domain_padding: Dict[str, List] = field(
+            default_factory=lambda: {"lower": [0, 0, 0], "upper": [0, 0, 0]}
+        )
+        parallel_k: Optional[bool] = None
+
+        def set_parallel_k(self, value: bool) -> "OirToNpir.Context":
+            return self.__class__(parallel_k=value, domain_padding=self.domain_padding)
+
+        def add_offset(self, axis: int, value: int) -> "OirToNpir.Context":
+            if value > 0:
+                current_value = self.domain_padding["upper"][axis]
+                self.domain_padding["upper"][axis] = max(current_value, value)
+            elif value < 0:
+                current_value = self.domain_padding["lower"][axis]
+                self.domain_padding["lower"][axis] = max(current_value, abs(value))
+            return self
+
+        def add_offsets(self, i: int, j: int, k: int) -> "OirToNpir.Context":
+            for axis_index, value in enumerate([i, j, k]):
+                self.add_offset(axis_index, value)
+            return self
+
+    def visit_Stencil(
+        self, node: oir.Stencil, *, ctx: Optional[Context] = None
+    ) -> npir.Computation:
+        ctx = ctx or self.Context()
+        vertical_passes = [self.visit(vloop, ctx=ctx) for vloop in node.vertical_loops]
+
+        return npir.Computation(
+            field_params=[decl.name for decl in node.params if isinstance(decl, oir.FieldDecl)],
+            params=[decl.name for decl in node.params],
```

Comment:
Do you have a strong opinion on this approach? Splitting scalar and non scalar params and throwing away ordering? We should do the same approach for all.

Reply:
where is ordering thrown away? The "params" used for the function signature completely preserves ordering of parameters. The "field_params" preserves the order of field params even though it doesn't matter in what it is used for.

Reply:
Ah sorry. I didn't check carefully. I thought it's a distinct set (like in definition ir).